asllib/Interpreter.ml

```diff
@@ -518,11 +531,12 @@ module Make (B : Backend.S) (C : Config) = struct
     let or' prev here = prev >>= B.binop BOR here in
     let rec in_values v ty =
       match (Types.get_structure (IEnv.to_static env) ty).desc with
-      | T_Real | T_Bool | T_Enum _ | T_String | T_Int None -> m_true
+      | T_Real | T_Bool | T_Enum _ | T_String | T_Int UnConstrained -> m_true
+      | T_Int (UnderConstrained _) -> fatal_from loc Error.TypeInferenceNeeded
```

Comment:
What does this error mean in this context?
I guess it's hard for me to follow the intention of the code. Some comments wouldn't hurt.